spec/requests/session/check/session_checks_controller_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Session::Check::SessionChecksController, type: :request do
+  describe "#time_to_session_expiry" do
+    it "returns the time to session expiry" do
+      expect(User.timeout_in).to eq(30.minutes)
+
+      Timecop.freeze("2023-01-02") do
+        last_request_at = Time.now.utc.to_i - 500
```

Comment:
Do we need to cast these all using `.to_i`? Can we not keep them in a time format? (and for example substract minutes/seconds as opposed to 500?

Reply:
You can't "a Time from seconds" it seems. But you can subtract "the number of seconds that a Time represents since the epoch" (which is an integer).

I've done a bit of work to show you what I can change, I don't know if it is an improvement.